exporters/zipkin/include/opentelemetry/exporters/zipkin/zipkin_exporter.h

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include "opentelemetry/exporters/zipkin/recordable.h"
+#include "opentelemetry/ext/http/client/http_client_factory.h"
+#include "opentelemetry/ext/http/common/url_parser.h"
+#include "opentelemetry/sdk/trace/exporter.h"
+#include "opentelemetry/sdk/trace/span_data.h"
+
+#include "nlohmann/json.hpp"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace zipkin
+{
+
+std::string ZIPKIN_ENDPOINT_DEFAULT = "http://localhost:9411/api/v2/spans";
+
+/**
+ * Struct to hold Zipkin  exporter options.
+ */
+struct ZipkinExporterOptions
+{
+  // The endpoint to export to. By default the OpenTelemetry Collector's default endpoint.
+  std::string endpoint     = ZIPKIN_ENDPOINT_DEFAULT;
+  TransportFormat format   = TransportFormat::JSON;
+  std::string service_name = "default-service";
+  std::string ipv4;
+  std::string ipv6;
+};
+
+namespace trace_sdk   = opentelemetry::sdk::trace;
+namespace http_client = opentelemetry::ext::http::client;
+
+/**
+ * The Zipkin exporter exports span data in JSON format as expected by Zipkin
+ */
+class ZipkinExporter final : public trace_sdk::SpanExporter, public http_client::EventHandler
+{
+public:
+  /**
+   * Create an ZipkinExporter using all default options.
+   */
+  ZipkinExporter();
+
+  /**
+   * Create an ZipkinExporter using the given options.
+   */
+  ZipkinExporter(const ZipkinExporterOptions &options);
```

Comment:
Use `explicit` for all 1 arguments constructor. I think it is a bad practice to rely on converting constructors (https://en.cppreference.com/w/cpp/language/converting_constructor) in general unless explicitly wanted to have this behavior like `std::string_view`.

/cc @maxgolov @lalitb @g-easy @jsuereth  I see that in other places you do not use `explicit`, what is the overall rule?

Reply:
Agree. Will fix it. We follow google c++ style guide:
(https://google.github.io/styleguide/cppguide.html#Implicit_Conversions):
`Do not define implicit conversions. Use the explicit keyword for conversion operators and single-argument constructors.`

Also, would be good to add `cpplint`` CI check to catch these error.

Reply:
This is fixed for zipkin. Will have separate PR for adding `explicit` keyword at other places.